frontend/lib/Repository/AdminRepository/GroupsRepository.dart

```diff
@@ -1,33 +1,40 @@
 import 'package:adpv_frontend/DataModels/GroupSummary.dart';
+import 'package:adpv_frontend/Repository/UserRepository/AuthGateway.dart';
+import 'package:adpv_frontend/Repository/UserRepository/UserGateway.dart';
 import 'package:dio/dio.dart';
 
 import '../../Common/URLs.dart';
 import '../../DataModels/BackendResponse.dart';
 
 class GroupsRepository {
   final Dio _client = Dio();
-  String token = 'todo: function obtaining toker';
+  UserGateway userGateway = UserGateway();
 
   Future<List<GroupSummary>> getGroupsSummary() async {
-    _client.options.headers["Authorization"] = "Bearer $token";
+    final AuthResponse authResponse = await userGateway.getFromMemory();
 
-    try {
-      final response = await _client.get(
-        backendURL + getGroupSummaryURL,
-      );
-      if (response.statusCode == 200) {
-        final BackendResponse backendResponse =
-            BackendResponse.fromJson(response.data);
-        if (backendResponse.error == "") {
-          List<GroupSummary> groupSummaryList = [];
-          groupSummaryList = backendResponse.data
-              .map<GroupSummary>((e) =>
-                  GroupSummary.fromJson(e)) // do not refactor! UFO MAGIC!
-              .toList();
-          return Future.value(groupSummaryList);
+    if (authResponse.success) {
```

Comment:
Czy jeżeli pobranie `authResponse` się nie powiedzie, to czy nie powinniśmy przejść do ekranu logowania? :thinking: Myślę, że to będzie do zrobienia w tasku z flow logowania [APDV-124](https://brzezins.atlassian.net/browse/APDV-124)